Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/tasks/presenter/TasksPresenterTest.kt

```diff
@@ -128,7 +185,25 @@ class TasksPresenterTest {
     fun given_ThePresenterIsPresenting_on_EmissionOfAnErrorEventWithNoData_it_ShouldPresentTheErrorScreen() {
         givenThePresenterIsPresenting()
 
-        tasksEventSubject.onNext(Event.error(Throwable()))
+        tasksEventSubject.onNext(Event.idle(noEmptyTasks()).asError(SyncError()))
+
+        Mockito.verify(loadingDisplayer).showErrorScreen()
+    }
+
+    @Test
+    fun given_ThePresenterIsPresenting_on_EmissionOfErrorEventWithSomeDeletedData_it_ShouldPresentErrorIndicator() {
+        givenThePresenterIsPresenting()
+
+        tasksEventSubject.onNext(Event.error<Tasks>(SyncError()).updateData(someDeletedLocallyTasks()))
```

Comment:
Done in https://github.com/novoda/spikes/pull/113/commits/dd2cab8371460ca792d7d5f06d27b5ca06fb819e